src/rdsk.c

```diff
@@ -280,54 +266,39 @@ struct RC_PROFILE *search_cache_targets(char *return_message)
 		print_error(msg, return_message, __func__, strerror(errno));
 		return NULL;
 	}
-	if ((num2 = scandir(SYS_BLOCK, &nodes, scandir_filter_dm, NULL)) < 0) {
-		msg = ERR_SCANDIR;
-		print_error(msg, return_message, __func__, strerror(errno));
-		list = clean_scandir(list, num);
-		return NULL;
-	}
 
-	for (;n < num; n++) {
+	for (n = 0; n < num; n++) {
 		prof = calloc(1, sizeof(struct RC_PROFILE));
 		if (prof == NULL) {
 			msg = ERR_CALLOC;
-			print_error(msg, return_message, __func__, strerror(errno));
-			list = clean_scandir(list, num);
-			nodes = clean_scandir(nodes, num2);
-			free_linked_lists(cache_head, NULL, NULL);
-			return NULL;
+			goto search_cache_targets_error;
 		}
 		strcpy(prof->device, list[n]->d_name);
+		sprintf(file, "%s/%s", DEV_MAPPER, list[n]->d_name);
+		readlink(file, link, sizeof(link));
+		memset(file, 0x0, sizeof(file));
+		sprintf(file, "%s/%s", SYS_BLOCK, link + 3);
+		if (access(file, F_OK) != SUCCESS) {
+			msg = "Error. Path %s does not exist.";
+			print_error(msg, return_message, file);
+			goto search_cache_targets_error2;
```

Comment:
`access()` call supports `errno`, so I believe you can use the same `goto` as the other error handling code blocks here too.